dev/format_code.sh

```diff
@@ -182,28 +186,35 @@ xmlsort_failed() {
   exit 7
 }
 echo "Formatting resource files with xmlsort..."
-if [ "$ALL" == "--all" ]; then
-  TRACKED=$(git ls-files -- "**.qrc")
-else
-  # Only files which differ from the master branch
-  TRACKED=$(git diff --name-only master -- "**.qrc")
-fi
-UNTRACKED=$(git ls-files --others --exclude-standard -- "**.qrc")
-for file in $TRACKED $UNTRACKED
+for dir in img/ libs/
 do
-  if [ -f "$file" ]; then
-    OLD_CONTENT=$(cat "$file")
+  for file in $(search_files "${dir}" .qrc)
+  do
     NEW_CONTENT=$(xmlsort -r "RCC/qresource/file" -i -s "$file" || xmlsort_failed)
     NEW_CONTENT="${NEW_CONTENT//\'/\"}"
-    if [ "$NEW_CONTENT" != "$OLD_CONTENT" ]
-    then
-      printf "%s\n" "$NEW_CONTENT" > "$file"
-      echo "[M] $file"
-      COUNTER=$((COUNTER+1))
-    else
-      echo "[ ] $file"
+    update_file "$file"
+  done
+done
+
+# Format qml source files with qmlformat
+qml_format_failed() {
+  echo "" >&2
+  echo "ERROR: qmlformat failed!" >&2
+  echo "  Make sure that qmlformat (qtdeclarative5-dev-tools) are installed." >&2
+  echo "  On Linux, you can also run this script in a docker" >&2
+  echo "  container by using the '--docker' argument." >&2
+  exit 7
+}
+echo "Formatting QML sources with qmlformat..."
+for dir in share/
```

Comment:
Same here, I wouldn't limit this to the share directory, there don't seem to be any third-party qml files in the repository.

Reply:
The problem is that in other case it formats the cmake-build-debug directory too and it is really huge as it contains a lot of file copies already changed. Mainly in CMakeFiles scripts but its same for other files. With this limit the speed of stylecheck/formater is comparably faster (0.5sec vs 30sec measured by eye)

Reply:
I can't reproduce this problem. The build directory created by qtcreator is located outside the source directory, thus it doesn't cause any troubles. And when building manually, the build directory should be named `build` to ensure it is ignored by Git (its *always* bad if the build directory is not ignored). With this directory ignored, `git ls-files` should not be slow.

Or am I misunderstanding something?

Note: If you prefer a different name for the build directory, you can add it to `.git/info/exclude` manually to make sure it is ignored as well.

Reply:
default build directory for CLion IDE is `cmake-build-debug` and `cmake-build-release` depending on the target, and it is in the root directory of the project. But yes, probably the `.git/info/exclude` solution will work for me. I will just try.

Reply:
Restriction removed